strongbox-web-core/src/main/java/org/carlspring/strongbox/interceptors/MavenArtifactRequestInterceptor.java

```diff
@@ -0,0 +1,138 @@
+package org.carlspring.strongbox.interceptors;
+
+import org.carlspring.strongbox.artifact.MavenArtifact;
+import org.carlspring.strongbox.artifact.MavenArtifactUtils;
+import org.carlspring.strongbox.controllers.layout.maven.MavenArtifactController;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.io.RepositoryPathResolver;
+import org.carlspring.strongbox.services.ConfigurationManagementService;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import static org.carlspring.strongbox.storage.metadata.MetadataHelper.MAVEN_METADATA_XML;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class MavenArtifactRequestInterceptor
+        extends HandlerInterceptorAdapter
+{
+
+    @Inject
+    private RepositoryPathResolver repositoryPathResolver;
+
+    @Inject
+    private ConfigurationManagementService configurationManagementService;
```

Comment:
@fuss86 I mean that we should use `StoragesConfigurationManager` here instead of `ConfigurationManagementService`. There is shortcut methods `StoragesConfigurationManager.getStorage` and `StoragesConfigurationManager.getRepository` instead of `ConfigurationManagementService.getConfiguration().getStorage(storageId)`.